infra/storage/shared_file_system_managers/kS3SharedFileSystemMgr.php

```diff
@@ -251,6 +252,20 @@ private function doPutFileHelper($filePath , $fileContent, $params)
 		catch (Exception $e)
 		{
 			KalturaLog::warning("Failed to uploaded to s3, info with message: " . $e->getMessage());
+			
+			if ($e instanceof S3MultipartUploadException)
```

Comment:
In this case of a resume do we still need the external retry which runs in the outer loop?

Reply:
Due to the way we wrote the S3 and OCI wrappers - we handle retries on the 'parent' function "doPutFileContent" or "doMoveLocalToShared" in that case.

If you call "doPutFileHelper" directly it will not handle the resume.
It must be called from the 'parent' function which will re-call "doPutFileHelper" after the exception with the added 'state' value.

To make this function 'independent' I needed to either implement an internal retry + a recursive call
Move the retries from the parents to "doPutFileHelper"
I feel it is not needed as the way we call "doPutFileHelper" is always from a parent that implement a retry.